### ISSUE TITLE
Why reduce mask in class-avg?

### ISSUE THREAD

**Issue by madoka109** [FlamieZhu/BCL#5]
Thanks for your work!
Here is a little question: why below equ. minus mask? mask is the count of positive pair samples, right?
per_ins_weight = torch.tensor([batch_cls_count[i] for i in targets], device=device).view(1, -1).expand(
            2 * batch_size, 2 * batch_size + len(self.cls_num_list)) - mask

**Comment by FlamieZhu** [comment]
Hi,
Note that eq. calculates the number of samples per class within a batch, while positives do not include the anchor itself, so there is a mask here.

**Comment by madoka109** [comment]
> 

I got it! Thanks a lot!

**Comment by sun-bowen** [comment]
I used to be confused about this, but the pieces fit together now.
